plutus-core/plutus-core/src/PlutusCore/TypeCheck/Internal.hs

```diff
@@ -509,6 +515,58 @@ inferTypeM (Error ann ty) = do
     checkKindM ann ty $ Type ()
     normalizeTypeM $ void ty
 
+-- resTy ~> vResTy     vResTy = sop s_0 ... s_i ... s_n     s_i = [p_0 ... p_m]
+-- [check| G !- t_j : p_j]
```

Comment:
I think I'd write the last line as

```
-- [check| G !- t_0 : p_0] ... [check| G !- t_m : p_m]
```

but whatever.

Reply:
It's awkward, I'm eliding a lot of iteration. If we start doing that then we should also include the iteration over the sums...

Reply:
I think I figured out a way to write it okay.